File: Algorithms/p055_Jump_Game/p055_Jump_Game_2.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        max_reachable = 0
        for i in range(len(nums)):
            if i>max_reachable:
                return False
            max_reachable = max(max_reachable,i+nums[i])
        return True

File: Algorithms/p055_Jump_Game/p055_Jump_Game_2.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_reachable = 0;
        for(int i=0;i<nums.size()&&i<=max_reachable&&max_reachable<nums.size()-1;++i){
            max_reachable = max(max_reachable,i+nums[i]);
        }
        return max_reachable>=nums.size()-1;
    }
};
